### PR TITLE
Add group membership filter widget in contact list

### LINKED ISSUE #8160: Filter contact list by their group membership
In the `/​contacts` page it would be nice to be able to fil­ter your con­tacts by group … I know there is a ‘Groups’ tab at the top, but that is for “organ­is­ing” and so it doesn’t have the same con­tex­tual options (for exam­ple: ‘View con­tact’ — which leads directly to the contact’s options area)

originally posted by Falgn0n The Wizard at [the support forum](https://librenet.co.za/display/f917d7dd-975e-29c6-e13b-b20523942976). See mocup there.

### PR DESCRIPTION
**Pull request by MrPetovan**
Closes #8160

### CODE REVIEW

File: src/Module/Contact.php [hunk 1]
```diff
@@ -759,14 +758,28 @@ public static function content(array $parameters = [], $update = 0)
...
+
 		$sql_extra .=  " AND NOT `deleted` ";
 
 		$sql_extra2 = ((($sort_type > 0) && ($sort_type <= Model\Contact::FRIEND)) ? sprintf(" AND `rel` = %d ", intval($sort_type)) : '');
 
 		$sql_extra3 = Widget::unavailableNetworks();
 
-		$r = q("SELECT COUNT(*) AS `total` FROM `contact`
-			WHERE `uid` = %d AND `self` = 0 $sql_extra $sql_extra2 $sql_extra3",
+		$r = q("SELECT COUNT(*) AS `total`
```
Comment (nupplaphil) [inline src/Module/Contact.php]: Maybe you can use `DBA::p()` so we lower the dependencies to `include\dba.php`

File: src/Module/Contact.php [hunk 2]
```diff
@@ -776,7 +789,17 @@ public static function content(array $parameters = [], $update = 0)
 
 		$contacts = [];
 
-		$r = q("SELECT * FROM `contact` WHERE `uid` = %d AND `self` = 0 $sql_extra $sql_extra2 $sql_extra3 ORDER BY `name` ASC LIMIT %d , %d ",
+		$r = q("SELECT contact.*
```
Comment (nupplaphil) [inline src/Module/Contact.php]: Maybe you can use `DBA::p()` so we lower the dependencies to `include\dba.php`

File: src/Module/Contact.php (hunk 1 above)
Comment (MrPetovan) [reply src/Module/Contact.php]: The whole damn thing needs to be rewritten and split between the contact list page and the single contact display page.

**Comment by MrPetovan** [comment]
Well, I certainly couldn't stop at `DBA::p` so I improved a bunch of other small stuff in the same file.

**Pull request merged**